src/commands/edge_app.rs

```diff
@@ -240,15 +234,23 @@ impl EdgeAppCommand {
         Ok(())
     }
 
-    pub fn upload(self, path: &Path) -> Result<(), CommandError> {
+    pub fn upload(self, path: &Path, app_id: Option<String>) -> Result<u32, CommandError> {
         let data = fs::read_to_string(path)?;
-        let manifest: EdgeAppManifest = serde_yaml::from_str(&data)?;
+        let mut manifest: EdgeAppManifest = serde_yaml::from_str(&data)?;
+
+        // override app_id if user passed it
+        if let Some(id) = app_id {
```

Comment:
Isn't it what get_actual_app_id does?

Reply:
Logic is different - we fail if there is no manifest because manifest also has settings